interactions/api/gateway/client.py

```diff
@@ -364,46 +364,41 @@ def _dispatch_event(self, event: str, data: dict) -> None:
                 if data["type"] == InteractionType.APPLICATION_COMMAND:
                     _name = f"command_{_context.data.name}"
 
-                    if _context.data._json.get("options"):
-                        for option in _context.data.options:
+                    if options := _context.data.options:
+                        for option in options:
                             _type = self.__option_type_context(
                                 _context,
-                                (option["type"] if isinstance(option, dict) else option.type.value),
+                                option.type.value,
                             )
                             if _type:
-                                if isinstance(option, dict):
-                                    _type[option["value"]]._client = self._http
-                                    option.update({"value": _type[option["value"]]})
-                                else:
-                                    _type[option.value]._client = self._http
-                                    option._json.update({"value": _type[option.value]})
+                                _type[option.value]._client = self._http
+                                # I'm not sure about line below because its doesn't affect anything
+                                option.value = _type[option.value]
```

Comment:
It should assign the resolved object to `option.value` so that it is not the ID of the object, if I'm reading it correctly